### DISCUSSION TITLE
Collapsing tree without using a parent tab?

### DISCUSSION THREAD

**Discussion by branables** [piroor/treestyletab#3419] [General]
Hi folks, 

I'm still pretty new to Tree Style Tabs, but I'm starting to get everything dialed in.
One thing I haven't found is the ability to collapse a tree (or even a sub tree) without having to go to collapse it from the parent tab. 

A good example is reddit comments. If there is a big thread, you can scroll down the page, click the indent line, and it will collapse everything in that "tree". This kind of functionality would be extremely helpful as I hoard a lot of tabs. My trees often are huge, and  I'd love to minimize scrolling. 

I've gone through the add ons and didn't see anything. I didn't see anything in the settings. I also didn't see anything similar in the discussions. 

Does this already exist? If not, I'll probably create this as a suggestion.

**Comment by piroor** [comment]
Go to addons manager page (about:addons), click the gear button at the header area, and choose "Manage Extension Shortcuts", then you can assign a custom shortcut for the feature "Collapse this Tree". Did you try it?

**Reply by branables** [reply]
Thanks for looking at this! Previously, I tried the "Collapse this Tree Recursively" and I just tried the "Collapse this Tree", but neither did what I want. The functionally I would love is to collapse the tab I'm currently in and automatically jump to the parent tab.

My goal it lessen the amount of scrolling I have to do as often times I'll have 100+ tabs connected to a parent and I want to jump to that parent tab by clicking next to the active tab. As mentioned, similar to how reddit will handle deep treads. 

Hopefully that makes sense? Thanks for your help! I love the extension!

**Reply by piroor** [reply]
Hmm, you looks to mean a command like "collapse parent tree". TST has no such built-in shortcut slot for that, so [TST More Tree Commands](https://github.com/piroor/tst-more-tree-commands) should cover it. I've created an issue for that: https://github.com/piroor/tst-more-tree-commands/issues/28

**Reply by branables** [reply]
This is perfect! I'll keep an eye out for the update! Thank you again!

**Reply by piroor** [reply]
Sorry I have less motivation to implement that by my hand, and pull requests to implement those "good first issue" labeled features are always welcome. I hope those issues get people involved to join opensource development...
